### PR TITLE
Update use of NonPortableCast to MemoryMarshal.Cast and cleanup

### PR DESCRIPTION
**Pull request by ahsonkhan**
Supersedes https://github.com/dotnet/corefxlab/pull/2089

Reacting to recent API changes from https://github.com/dotnet/corefx/pull/26467

cc @dotnet/corefxlab-contrib

### CODE REVIEW

**Comment by joshfree** [comment]
I assume VS generated all the formatting changes and this wasn't done by hand? 😄

**Comment by ahsonkhan** [comment]
@dotnet-bot test Innerloop OSX10.12 Release Build and Test
@dotnet-bot test Innerloop Ubuntu16.04 Debug Build and Test

**Comment by ahsonkhan** [comment]
> I assume VS generated all the formatting changes and this wasn't done by hand?

Yes :)

**Comment by joshfree** [comment]
Also, could you create tracking issues for any intermittent test failures in CI?

File: tests/System.Buffers.Primitives.Tests/CastTests.cs [hunk 1]
```diff
@@ -56,7 +58,7 @@ public void IntArraySpanCastedToByteArraySpanHasSameBytesAsOriginalArray()
             var ints = new int[100000];
             Random r = new Random(42324232);
             for (int i = 0; i < ints.Length; i++) { ints[i] = r.Next(); }
-            var bytes = ints.AsSpan().NonPortableCast<int, byte>();
+            var bytes = MemoryMarshal.Cast<int, byte>(ints.AsSpan());
```
Comment (KrzysztofCwalina) [inline tests/System.Buffers.Primitives.Tests/CastTests.cs]: Why are you changing it to Cast and not to TryCast?

Comment (ahsonkhan) [reply tests/System.Buffers.Primitives.Tests/CastTests.cs]: It wasn't TryCast before. And we don't have that yet.

**Comment by ahsonkhan** [comment]
> Also, could you create tracking issues for any intermittent test failures in CI?

The failures are coming from the already known issue: https://github.com/dotnet/corefxlab/issues/2062

**Pull request merged**